src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,148 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import core.basesyntax.service.UserDataException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String VALID_PASSWORD = "VALID_PASSWORD";
+    private static final String INVALID_SHORT_PASSWORD = "ISP";
+    private static final Integer VALID_AGE = 30;
+    private static final Integer MIN_VALID_AGE = 18;
+    private static final Integer MIN_INVALID_AGE = 16;
+    private static final Integer NEGATIVE_INVALID_AGE = -10;
+    private static final Integer INVALID_LARGE_AGE = 130;
+    private static final String VALID_LOGIN = "VALID_LOGIN";
```

Comment:
In your test classes that is OK if you use literals directly instead of constants. But, if you already have created a lot of constants, you can move them to a separate class, e.g. RegistrationServiceTestUtils or RegistrationServiceTestConstants